core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/add-answer-group-modal.controller.ts

```diff
@@ -51,10 +51,15 @@ angular.module('oppia').controller('AddAnswerGroupModalController', [
       $uibModalInstance: $uibModalInstance
     });
     const eventBusGroup: EventBusGroup = new EventBusGroup(EventBusService);
+    $scope.modalId = Symbol();
     $scope.isInvalid = false;
     eventBusGroup.on(
       ObjectFormValidityChangeEvent,
-      event => $scope.isInvalid = event.message.value);
+      event => {
+        if (event.message.modalId === $scope.modalId) {
+          $scope.isInvalid = event.message.value;
```

Comment:
is an applyAsync needed here?

Reply:
Yes... I noticed sometimes the button didn't get disabled without it.

Reply:
Cool, please add one. Thanks!

Reply:
Added!